### PR TITLE
BigTree SKR Pro V1.1 board support

### PR DESCRIPTION
**Pull request by Msq001**
### Requirements

* add support BigTree SKR Pro V1.1 board

### Description

1. add `BIGTREE_GENERIC_STM32F407_5X` variant, this variant is a gerneric variant of STM32F405/407, include "F405Rx(64pins), F405/407Vx(100pins), F405/407Zx(144pins), F407Ix(176pins)", but the `ldscript.ld` is not gerneric, this .ld set the offset address of flash from 0x08000000 to 0x08008000(board bootloader jump address).
2. redefine `pin_t` from `int8_t` to `int16_t`, because of STM32F407Ix has up to 140 gpios, Overflow range of `int8_t`

### Related Issues
#14484 for this feature request

### CODE REVIEW

File: Marlin/src/pins/pins_BIGTREE_SKR_PRO_V1.1.h [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+#endif
+
+//
+// Temperature Sensors
+//
+#define TEMP_0_PIN         PF3  // T0
+#define TEMP_1_PIN         PF4  // T1
+#define TEMP_2_PIN         PF5  // T2
+#define TEMP_BED_PIN       PF6  // TB
+
```
Comment (thisiskeithb) [inline Marlin/src/pins/pins_BIGTREE_SKR_PRO_V1.1.h:160]: Can you confirm these are the correct pins? They don't match the documentation & there's at least one report in the BigTreeTech Facebook group with the correct pins being:

```c++
#define TEMP_BED_PIN       PF3  // T0 <=> TB
#define TEMP_0_PIN         PF4  // T1
#define TEMP_1_PIN         PF5  // T2
#define TEMP_2_PIN         PF6  // T3
```

I'm still waiting on my board to arrive, so I can't confirm.

**Comment by Msq001** [comment]
oh. onboard string is T0 T1 T2 T3.  hearter string is HEAT0 HEAT1 HEAT2 HOT_BED， I think T0 for HEAT0, T1 for HEAT1, T2 for HEAT2, and T3 for HOTBED is better.

**Comment by davidbilly** [comment]
@Msq001 hi, when this can merge with the latest marlin source, or I need to get from here : https://github.com/bigtreetech/BIGTREETECH-SKR-PRO-V1.1/tree/master/firmware/Marlin-SKR-Pro/Marlin . is this up to date with marlin code?

**Comment by Msq001** [comment]
hi. this two branch is same code for skr pro

File: Marlin/src/HAL/HAL_STM32/HAL.h [hunk 2]
```diff
@@ -132,7 +132,7 @@
 // Types
 // --------------------------------------------------------------------------
 
-typedef int8_t pin_t;
+typedef int16_t pin_t;
```
Comment (thinkyhead) [inline Marlin/src/HAL/HAL_STM32/HAL.h:135]: Do you suppose this is okay for all STM32 platforms, or would some be okay with 8 bits? Is STM32 one of those MCUs where 16 bit integers are more optimal, so it's no difference to use this for all of them?

**Comment by ShaggyDog18** [comment]
tested the recent 6-JUL release: 12864 display is not working properly, too many artifacts on the screen...Kind of synchronization issue.
So, something in the display interface should be tuned up...

**Comment by thinkyhead** [comment]
> So, something in the display interface should be tuned up...

Keep us posted if you figure anything out. Display experimenters with SKRs are hard to find….

**Comment by thisiskeithb** [comment]
Looks like there's still some work being done: https://github.com/bigtreetech/BIGTREETECH-SKR-PRO-V1.1/issues/8

**Pull request merged**